Build fails when autoconvert feature is disabled

The new `impl_from!` macro unconditionally uses `si::change_base`, which is only available when the `autoconvert` feature is enabled. Disabling the feature causes the following build error:
```
error[E0432]: unresolved import `si::change_base`
  --> src\si\mod.rs:93:14
   |
93 |     use si::{change_base, Dimension, Quantity, Units};
   |              ^^^^^^^^^^^ no `change_base` in `si`
```

This is a regression in uom 0.23.0, introduced in #132.

--- a/.travis.yml
+++ b/.travis.yml
@@ -27,6 +27,7 @@ matrix:
     script: |
       set -e
       cargo test --all --verbose --features "use_serde"
+      cargo test --verbose --no-default-features --features "f32 si"
       cargo test --verbose --no-default-features --features "autoconvert f32 si use_serde"
       cargo test --verbose --no-run --no-default-features --features "autoconvert usize isize bigint bigrational si std use_serde"
       cargo test --verbose --no-run --no-default-features --features "autoconvert usize u8 u16 u32 u64 isize i8 i16 i32 i64 bigint biguint rational rational32 rational64 bigrational f32 f64 std use_serde"

--- a/src/si/mod.rs
+++ b/src/si/mod.rs
@@ -90,7 +90,7 @@ storage_types! {
 
 /// Primitive traits and types representing basic properties of types specific to the SI.
 pub mod marker {
-    use si::{change_base, Dimension, Quantity, Units};
+    use si::{Dimension, Quantity, Units};
     use Kind;
 
     /// AngleKind is a `Kind` for separating angular quantities from their indentically dimensioned
@@ -118,6 +118,7 @@ pub mod marker {
     }
 
     /// `impl_from` generates generic inter-Kind implementations of `From`.
+    #[cfg(feature = "autoconvert")]
     #[macro_export]
     macro_rules! impl_from {
         ($a:ident, $b:ident) => {
@@ -153,12 +154,55 @@ pub mod marker {
                     Self {
                         dimension: ::lib::marker::PhantomData,
                         units: ::lib::marker::PhantomData,
-                        value: change_base::<
+                        value: super::change_base::<
                             Dimension<L = L, M = M, T = T, I = I, Th = Th, N = N, J = J, Kind = $b>,
                             Ul,
                             Ur,
                             V,
                         >(&val.value),
+                    }
+                }
+            }
+        };
+    }
+
+    /// `impl_from` generates generic inter-Kind implementations of `From`.
+    #[cfg(not(feature = "autoconvert"))]
+    #[macro_export]
+    macro_rules! impl_from {
+        ($a:ident, $b:ident) => {
+            impl<L, M, T, I, Th, N, J, U, V>
+                From<
+                    Quantity<
+                        Dimension<L = L, M = M, T = T, I = I, Th = Th, N = N, J = J, Kind = $a>,
+                        U,
+                        V,
+                    >,
+                >
+                for Quantity<
+                    Dimension<L = L, M = M, T = T, I = I, Th = Th, N = N, J = J, Kind = $b>,
+                    U,
+                    V,
+                >
+            where
+                U: Units<V> + ?Sized,
+                V: ::num_traits::Num + ::Conversion<V>,
+            {
+                fn from(
+                    val: Quantity<
+                        Dimension<L = L, M = M, T = T, I = I, Th = Th, N = N, J = J, Kind = $a>,
+                        U,
+                        V,
+                    >,
+                ) -> Quantity<
+                    Dimension<L = L, M = M, T = T, I = I, Th = Th, N = N, J = J, Kind = $b>,
+                    U,
+                    V,
+                > {
+                    Self {
+                        dimension: ::lib::marker::PhantomData,
+                        units: ::lib::marker::PhantomData,
+                        value: val.value,
                     }
                 }
             }

